src/test/scala/mesosphere/marathon/storage/repository/GcActorTest.scala

```diff
@@ -618,5 +619,40 @@ class GcActorTest extends AkkaUnitTest with TestKitBase with GivenWhenThen with
         noMoreInteractions(deployRepo)
       }
     }
+    "starting up with provided scan interval" should {
+      "start in a resing state" in {
+        val appRepo = mock[AppRepositoryImpl[RamId, String, Identity]]
+        val podRepo = mock[PodRepositoryImpl[RamId, String, Identity]]
+        val groupRepo = mock[StoredGroupRepositoryImpl[RamId, String, Identity]]
+        val deployRepo = mock[DeploymentRepositoryImpl[RamId, String, Identity]]
+        val actor = TestFSMRef(new GcActor(deployRepo, groupRepo, appRepo, podRepo, 25, 32, 100.millis))
+
+        actor.stateName shouldEqual Resting
+      }
+      "go to Idle when the resing time is over" in {
+        val appRepo = mock[AppRepositoryImpl[RamId, String, Identity]]
+        val podRepo = mock[PodRepositoryImpl[RamId, String, Identity]]
+        val groupRepo = mock[StoredGroupRepositoryImpl[RamId, String, Identity]]
+        val deployRepo = mock[DeploymentRepositoryImpl[RamId, String, Identity]]
+        val actor = TestFSMRef(new GcActor(deployRepo, groupRepo, appRepo, podRepo, 25, 32, 100.millis))
+
+        actor.stateName shouldEqual Resting
+        actor.isTimerActive(RestingTime) shouldEqual true
+        Thread.sleep(150)
```

Comment:
This is an anti-pattern. Use `eventually` with a defined patienceConfig (if necessary) instead.